[Feat] 이벤트 생성 화면 구현

serena0720 opened the pull request:
## 📌 Related Issue
- #43 

## 🚀 Description
- EventStart, EventProcess 화면 구현하였습니다.
- 관련 화면을 CreatEventCoordinator로 연결하였습니다.

## ✅ Done
- EventStart 화면에 로띠파일 넣어 생성하였습니다.
- EventProcess 화면에 텍스트뷰, CurrentDate, ImagePicker 화면을 구현하였습니다.
- Design파일 : GabbangzipDate, SelectPhoto를 생성하였습니다.
- CreateEventCoordinator를 생성하였습니다.

## 📸 Screenshot
![2](https://github.com/user-attachments/assets/964ec96f-d300-4aed-b68e-48d70067f921) ![1](https://github.com/user-attachments/assets/95bde822-0f5f-46cd-8af5-43696d9624da)


## 📢 Notes
 - Event API는 별도 이슈 생성하여 작업 예정입니다.🙇‍♀️
 - CreateEventCoordinator가 MainCoordinator를 알게되면 순환구조 문제가 발생하는데, 이를 어떻게 해결해야할지 모르겠습니다.🤔 현재는 CreatEventCoordinator에서 GroupList를 띄우는 방식으로 구현하였습니다. 하지만 이렇게되면 계속 뷰가 쌓이기 때문에 좋은 방식이 아닌거 같아 고민이 됩니다..
 > [MainCoordinator, [CreateEventCoordinator]] 이런 방식으로 쌓일거라 CreateEventCoordinator 전체를 pop 혹은 dismiss를 시키면 될거라 생각했습니다. presentcover로 CreateEventCoordinator를 올렸으니 dismissAll / dismiss를 하면 될 거라 생각했는데, 그냥 CreateEventCoordinator의 Root로 이동합니다.. 제가 어떤 부분을 놓치는걸까요🤦‍♀️
 - 현재 CreateEventCoordinator를 삭제해야할 수도 있습니다.. 이벤트 생성 Start 화면이 불필요하다는 코멘트가 있었습니다.🥹
 이 부분 상의되는 대로 업데이트 공유하겠습니다. 오히려 MyPageCoordinator를 살리는것이 현명한 판단이었을 거 같습니다.. 역시 대장님의 혜안..
- route 부분의 코드길이가 길어, 컨벤션 회의가 필요할 거 같습니다 @zekexros @hope1053 @GREENOVER

serena0720 committed:
feat: CreateEventCoordinator, CreateEvent tuist Dependency 생성

serena0720 committed:
feat: CreateEventCoordinator, CreateEvent tuist Dependency 생성

serena0720 committed:
feat: CreateEventCoordinator, CreateEvent tuist Dependency 생성

serena0720 committed:
feat: CreateEventCoordinator, CreateEvent tuist Dependency 생성

serena0720 committed:
feat: CreateEvent, CreateEventCoordinator

serena0720 committed:
feat: CreateEvent, CreateEventCoordinator

serena0720 committed:
feat: CreateEvent, CreateEventCoordinator

serena0720 committed:
feat: CreateEvent, CreateEventCoordinator

serena0720 committed:
feat: CreateEvent, CreateEventCoordinator

serena0720 committed:
feat: CreateEvent, CreateEventCoordinator

serena0720 committed:
feat: CreateEventStartView, CreateEventStartCore 셍성

serena0720 committed:
feat: CreateEventStartView, CreateEventStartCore 셍성

serena0720 committed:
feat: CreateEventStartView, CreateEventStartCore 셍성

serena0720 committed:
feat: CreateEventStartView, CreateEventStartCore 셍성

serena0720 committed:
feat: CreateEventStartView, CreateEventStartCore 셍성

serena0720 committed:
feat: CreateEventStartView, CreateEventStartCore 셍성

serena0720 committed:
Merge branch 'feat/#43' of https://github.com/mash-up-kr/gabbangzip-iOS into feat/#43

# Conflicts:
#	Projects/Features/Coordinator/MainCoordinator/MainCoordinatorView.swift
#	Projects/Features/Coordinator/MainCoordinator/MainScreen.swift
#	Projects/Features/Coordinator/Project.swift
#	Tuist/ProjectDescriptionHelpers/TargetDependency+extensions.swift

serena0720 committed:
feat: CreateEventProcessCore, CreateEventProcessView

serena0720 committed:
feat: CreateEventProcessCore, CreateEventProcessView

serena0720 committed:
feat: CreateEventProcessCore, CreateEventProcessView

serena0720 committed:
feat: CreateEventProcessCore, CreateEventProcessView

serena0720 committed:
chore: Rebase 코드 수정 및 오타 수정

serena0720 committed:
chore: Rebase 코드 수정 및 오타 수정

serena0720 committed:
chore: Rebase 코드 수정 및 오타 수정

serena0720 committed:
chore: Rebase 코드 수정 및 오타 수정

serena0720 committed:
Design: GabbangzipDate 생성

serena0720 committed:
Design: GabbangzipDate 생성

serena0720 committed:
Design: GabbangzipDate 생성

serena0720 committed:
feat: 현재 날짜 로직 추가

serena0720 committed:
feat: 현재 날짜 로직 추가

serena0720 committed:
feat: 현재 날짜 로직 추가

serena0720 committed:
feat: 현재 날짜 로직 추가

serena0720 committed:
feat: GroupList에서 이벤트 생성하기 클릭시 CreateEventCoordinator 연결

serena0720 committed:
feat: GroupList에서 이벤트 생성하기 클릭시 CreateEventCoordinator 연결

serena0720 committed:
feat: GroupList에서 이벤트 생성하기 클릭시 CreateEventCoordinator 연결

serena0720 committed:
feat: GroupList에서 이벤트 생성하기 클릭시 CreateEventCoordinator 연결

serena0720 committed:
refactor: MemberListView 접근제어자 수정

serena0720 committed:
refactor: MemberListView 접근제어자 수정

serena0720 committed:
refactor: MemberListView 접근제어자 수정

serena0720 committed:
refactor: MemberListView 접근제어자 수정

serena0720 committed:
feat: CreateEventCoordinator case 추가

serena0720 committed:
feat: CreateEventCoordinator case 추가

serena0720 committed:
feat: CreateEventCoordinator case 추가

serena0720 committed:
feat: CreateEventCoordinator case 추가

serena0720 committed:
refactor: MemberListCore init 추가

serena0720 committed:
refactor: MemberListCore init 추가

serena0720 committed:
chore: Event Picture 추가

serena0720 committed:
chore: Event Picture 추가

serena0720 committed:
chore: Event Picture 추가

serena0720 committed:
feat: 사진 선택 기능 추가 및 이벤트 만들기 화면 이동 추가

serena0720 committed:
feat: 사진 선택 기능 추가 및 이벤트 만들기 화면 이동 추가

serena0720 committed:
feat: 사진 선택 기능 추가 및 이벤트 만들기 화면 이동 추가

serena0720 committed:
feat: 사진 선택 기능 추가 및 이벤트 만들기 화면 이동 추가

serena0720 committed:
refactor: CreateEventProcess 뷰 수정

serena0720 committed:
refactor: CreateEventProcess 뷰 수정

serena0720 committed:
refactor: CreateEventProcess 뷰 수정

serena0720 committed:
refactor: CreateEventProcess 뷰 수정

serena0720 committed:
feat: CreateEventProcess completeButton 조건 추기

serena0720 committed:
feat: CreateEventProcess completeButton 조건 추기

serena0720 committed:
feat: CreateEventProcess completeButton 조건 추기

serena0720 committed:
feat: CreateEventProcess completeButton 조건 추기

serena0720 committed:
feat: CreateEventCoordinator 설정

serena0720 committed:
feat: CreateEventCoordinator 설정

serena0720 committed:
feat: CreateEventCoordinator 설정

serena0720 committed:
feat: CreateEventCoordinator 설정

serena0720 committed:
chore: 불필요한 주석 삭제

serena0720 committed:
chore: 불필요한 주석 삭제

serena0720 committed:
chore: 불필요한 주석 삭제

serena0720 committed:
chore: 불필요한 주석 삭제

GREENOVER commented:
@serena0720 
리뷰하기 앞서 이상한게 있어서 말씀드려요.
첫번째 gif를 보니까 이벤트 만들기 뷰에서 뒤로 가기를 통해 홈 메인 화면이 나오는데, 왜 push되는 그림이죠?
(우측에서 슬라이딩해서 나오는 전형적인 push)
코멘트에 남겨주셨던, CreateEventCoordinator가 MainCoordinator를 알고 있어서 push 시키고 있나요?
그렇다면 역전이 된거에요.
MainCoordinator가 CreateEventCoordinator를 아는 구조거나 아니면 MainCoordinator와 CreateEventCoordinator 같은 레벨로 본다면 그 두개를 가지고 있는 상위 코디네이터 route 액션에서 pop을 발행해주는 구조로 가야합니다.
좋은 방식이 아닌걸 떠나서, 아예 잘못된 흐름이기에 수정을 요청드립니다 🙏🏻
즉, [MainCoordinator, [CreateEventCoordinator]] 이런 방식으로 쌓이는것이 흐름상 맞다면 같은 레벨 선상으로 볼텐데 그럼 이걸 관장하는 app이나 home 코디네이터 (상위 코디네이터 객체)에서 route를 다뤄줘야 의도하신 CreateEventCoordinator 스택을 pop했을때 MainCoordinator가 적절히 나오고 할것 같아요.
여튼... 아래 리뷰 반영보다 우선적으로 요 부분 긴급 수정해주셔야 될듯 합니다.

GREENOVER commented on Projects/DesignSystem/Sources/Views/Date/GabbangzipDate.swift:
```diff
@@ -0,0 +1,44 @@
+public struct GabbangzipDate: View {
+  private var date: String
+  
+  public init(
+    date: String
+  ) {
```
한줄이여도 충분할것 같아요~!

serena0720 replied:
앗 거의 기계적으로 쳐놨더니...! 꼼꼼한 리뷰 감사합니다 유사 케이스 함께 수정했습니다!

GREENOVER commented on Projects/DesignSystem/Resources/Icons.xcassets/Event/delete.imageset/Contents.json:
```diff
@@ -0,0 +1,21 @@
+{
+  "images" : [
```
svg 추가하신 아이콘들 다 1배수만 적용되있죠~?
우측 인스펙터 섹션에서 scales를 single scales로 바꿔주세요~
(작업으로 추가하신 svg 전부요.)

serena0720 replied:
와 또 하나 배웠습니다 감사합니다!

GREENOVER commented on Projects/DesignSystem/Sources/Views/PhotoPicker/PhotoInfo.swift:
```diff
@@ -8,7 +8,7 @@
 import Foundation
 
-public struct PhotoInfo: Equatable {
+public struct PhotoInfo: Equatable, Hashable {
```
Hashable 필요한 이유 설명해주실수 있나요~?

GREENOVER replied:
추가로, 이건 플젝을 떠나서 생각해보시면 좋을것 같아요.
Equatable은 무엇이고 Hashable은 기본적으로 뭘 채택하고 있을까요~?
두 쓰임이 정확히 어떤것이고 우리 플젝 현재 코드에서는 어떤것을 채택하는게 왜 적절한지 서술해봐주세요 🙇🏻

serena0720 replied:
Equatable은 두개의 값이 동일한지를 비교, Hashable을 Key값을 Hashtable을 통해 Hashvalue를 찾아 비교합니다.
이때 Hashable이 Equatable을 상속하고 있기 때문에 Hashable을 사용한다면 Equatable을 채택하지 않아도 될 거 같습니다!

현재 코드에선 이미지의 고유한 key값으로 해시값을 비교하여 동일한 사진 중복을 막고자 하여 Hashable을 채택하였습니다.

GREENOVER replied:
@serena0720 
동일 사진 중복을 막고자한다면, Equatable만으로 충분하지 않나요?
Hashable은 어느정도 이해하고 계시지만, 놓친 부분이 해시 값을 사용하는 목적에서 놓쳤다고 생각해요.
해시 값은 동등성 비교를 위해 직접 사용이 되는것의 목적보다 빠른 검색에 있어요.
즉, 딕셔너리나 Set 같은 친구들에서 순환하며 요소로 판단될때 사용한것이죠.
그렇기에 현재 코드에서 단순 동일 사진 중복 체크를 위함이라면 Equatable만으로 충분합니다.

GREENOVER commented on Projects/DesignSystem/Sources/Views/SelectPhoto/SelectPhoto.swift:
```diff
@@ -0,0 +1,57 @@
+        HStack(spacing: 0) {
+          Text("\(selectedPhotosInfo.count)")
+            .font(.body16)
+            .foregroundStyle(selectedPhotosInfo.isEmpty ? DesignSystem.Colors.gray60 : DesignSystem.Colors.gray80)
+          Text("/\(maxCount)")
```
적절히 한줄 개행해보기ㅎㅎ

serena0720 replied:
넵 수정하겠씁니다!

GREENOVER commented on Projects/DesignSystem/Sources/Views/SelectPhoto/SelectPhoto.swift:
```diff
@@ -0,0 +1,57 @@
+  }
+  
+  public var body: some View {
+    HStack(spacing: 0) {
```
왜 HStack으로 한번 더 감쌌어요?
불필요합니다.
제거하고 적절히 나타나도록 수정해주세요 🙏🏻

serena0720 replied:
오우 감사합니다!!

GREENOVER commented on Projects/Features/Coordinator/CreateEventCoordinator/CreateEventCoordinatorCore.swift:
```diff
@@ -0,0 +1,58 @@
+  public var body: some Reducer<State, Action> {
+    Reduce { state, action in
+      switch action {
+      case .router(.routeAction(id: _, action: .createEventStart(.moveToCreateEventProcess))):
```
저는 보통 저희 Lint룰로 정한것처럼 인자 기준으로 개행을 하긴 하는데 요건 뭐 스타일이라 PR 코멘트에 적어주신것처럼 같이 토의해보시면 될것 같아요ㅎㅎ

GREENOVER commented on Projects/Features/Coordinator/CreateEventCoordinator/CreateEventCoordinatorView.swift:
```diff
@@ -0,0 +1,40 @@
+  }
+  
+  public var body: some View {
+    TCARouter(store.scope(state: \.routes, action: \.router)) { screen in
+      Group {
+        switch screen.case {
```
CreateEvent쪽에서 흐름이 단방향이 아닌가요?
즉, 탭바처럼 스위칭해서 독립적으로 뷰가 보여지는 구조인지 몰라서요~!
제가 코드만봐서 모를수 있는데, 이전 PR 코멘트에 남긴것처럼 해당 코디네이터 구조 리팩토링이 필요해서 일단 크게 신경안쓰고 넘어가겠습니다~
추후, 개선되면 그때 리뷰하겠습니다.

serena0720 replied:
현재 화면이 단일뷰로 수정되면서 CreateEventCoordinator를 삭제하기로 하였습니다🥹 이와 관련된 이슈들 모두 함께 닫도록 하겠습니다🙇‍♀️

GREENOVER commented on Projects/Features/Scene/CreateEventScene/CreateEventProcess/CreateEventProcessCore.swift:
```diff
@@ -0,0 +1,163 @@
+    case popupRightButtonTapped
+    case completeButtonTapped
+    case deleteSelectedPhoto(Int)
+    case onAppear
```
순서상 항상 제일먼저 타니 앞으로 가도될것 같아요~
(작은 의견)

GREENOVER commented on Projects/Features/Scene/CreateEventScene/CreateEventProcess/CreateEventProcessCore.swift:
```diff
@@ -0,0 +1,163 @@
+        state.text = text
+        
+        return .run { [state] send in
+          if state.text.isEmpty {
+            await send(.changeIsEventNamedStatus(false))
+          } else {
+            await send(.changeIsEventNamedStatus(true))
+          }
```
```suggestion
          await send(.changeIsEventNamedStatus(state.text.isEmpty))
```
한줄로 표현하면 간결할것 같아요.

serena0720 replied:
와 너무 좋은 조언 감사합니다!!

GREENOVER commented on Projects/Features/Scene/CreateEventScene/CreateEventProcess/CreateEventProcessCore.swift:
```diff
@@ -0,0 +1,163 @@
+        
+      case .onAppear:
+        let currentDate = {
+          let formatter = DateFormatter()
+          formatter.dateFormat = "YY/MM/dd"
+          return formatter.string(from: Date())
```
DateFormatter+extensions에 확장시켜서 사용해주세요~
일원화 관리하는게 편합니다.
필요하면 String+extensions도 보시구요~!

serena0720 replied:
GroupCore와 동일하게 DateFormatter를 활용하며, RecentEvent를 받아서 처리하는 방식으로 수정하겠습니다!

GREENOVER commented on Projects/Features/Scene/CreateEventScene/CreateEventProcess/CreateEventProcessCore.swift:
```diff
@@ -0,0 +1,163 @@
+        }
+        
+      case .changeIsPhotoSelected:
+        if state.selectedPhotosInfo.count == 4 {
+          state.isPhotoSelected = true
+        } else {
+          state.isPhotoSelected = false
+        }
```
코드를 줄여보세요~

serena0720 replied:
넹~ 삼항으로 변경했습니다

GREENOVER commented on Projects/Features/Scene/CreateEventScene/CreateEventStart/CreateEventStartCore.swift:
```diff
@@ -0,0 +1,61 @@
+  public struct State: Equatable {
+    public var groupName: String
+    
+    public init(
+      groupName: String = ""
+    ) {
```
파라미터 1개 = 한줄 (우리의 코드 컨벤션)

GREENOVER commented on Projects/Features/Scene/CreateEventScene/CreateEventProcess/CreateEventProcessView.swift:
```diff
@@ -0,0 +1,178 @@
+  }
+  
+  public var body: some View {
+    VStack(spacing: 0) {
```
뷰 비대합니다.
나누주세요 🙏🏻

GREENOVER replied:
그리고 되게 VStack으로 여러 중첩되어 쌓여있는데, 점검해주세요.
필요없는 Stack 제거해주세요ㅠ
만약 이렇게 꼭 표현되어야 한다?
그럼 코드가 잘못된걸거에요.. 다른 방법을 모색해보셔야 합니다.

GREENOVER commented on Projects/Features/Scene/CreateEventScene/CreateEventProcess/CreateEventProcessView.swift:
```diff
@@ -0,0 +1,178 @@
+          backButtonAction: { store.send(.backButtonTapped) }
+        )
+        
+        VStack(spacing: 0) {
```
VStack에서 또 아래 VStack으로 감쌀 필요가 있나요?
이미 수직적 계층으로 쌓이고 있는 형태로 보여서요.

GREENOVER commented on Projects/Features/Scene/CreateEventScene/CreateEventProcess/CreateEventProcessView.swift:
```diff
@@ -0,0 +1,178 @@
+          if !$store.selectedPhotosInfo.isEmpty {
+            ScrollView(.horizontal) {
+              HStack(spacing: 0) {
+              GabbangzipPhotoPicker(
```
Indent 안맞아요.

serena0720 replied:
헉 감사합니다!

GREENOVER commented on Projects/Features/Scene/CreateEventScene/CreateEventProcess/CreateEventProcessView.swift:
```diff
@@ -0,0 +1,178 @@
+              HStack(spacing: 0) {
+              GabbangzipPhotoPicker(
+                selectedPhotosInfo: $store.selectedPhotosInfo.sending(\.selectedImagesChanged),
+                isPresentedError: .constant(false),
```
에러는 표시안해서 상수처럼 박아두신거죠~?
쓸거면 수정해주시고 값 넣어줄 필요없어요.
해당 컴포넌트 들어가보시면 기본값 들어가있습니다.

serena0720 replied:
SelectGroupPhotoView에서도 사용 시 명시했길래 통일되게 한것인데 삭제하도록하겠습니다!

GREENOVER commented on Projects/Features/Scene/CreateEventScene/CreateEventProcess/CreateEventProcessView.swift:
```diff
@@ -0,0 +1,178 @@
+                              .frame(width: 26, height: 26)
+                          }
+                        )
+                        .padding([.trailing], -8)
```
하나인데 배열에 담아줄 필요는 없어보이네요!

serena0720 replied:
앗... 이래저래 고민하다가 놓쳤네요ㅜ 감사합니다!

GREENOVER commented on Projects/Features/Scene/CreateEventScene/CreateEventProcess/CreateEventProcessView.swift:
```diff
@@ -0,0 +1,178 @@
+                }
+              }
+            }
+          } else {
```
위 코멘트에서도 말했듯, 무조건 뷰 쪼개주세요.
비대합니다.

GREENOVER commented on Projects/Features/Scene/CreateEventScene/CreateEventStart/CreateEventStartView.swift:
```diff
@@ -0,0 +1,80 @@
+          type: .morphing,
+          loopMode: .loop
+        )
+        .frame(width: geometry.size.width - 229, height: geometry.size.width - 229)
```
229 이건 DS 상 수치인거죠?ㅋㅋ
frame으로 이렇게 줄거면, 이걸 생각해보셔야 합니다.
작은 디바이스에서는 정말 작게 보일거고,
큰 디바이스에서는 비교적 크게 보일거에요.
즉, 화면에 대해 해당 로티 비율이 다를겁니다 디바이스마다
그래서 비율 유지시킬거면 패딩으로~

serena0720 replied:
해당 뷰가 삭제되면서 전체적인 수정하도록 하겠습니다🥹

GREENOVER reviewed (changes requested):
일단 좀 문제되는 부분만 핵심으로 남겼어요~
전체 점검 하셔야 합니다!

zekexros commented on Projects/DesignSystem/Sources/Views/Date/GabbangzipDate.swift:
```diff
@@ -0,0 +1,44 @@
+      Spacer()
+    }
+    .background(DesignSystem.Colors.gray40)
+    .cornerRadius(10)
```
P4:
피그마에선 Radius가 8로 설정되어있는데 혹시 다시 한번 확인해주시겠어요?!

serena0720 replied:
해당 뷰와 동일한 모양으로 가는 "이벤트 한줄 요약"의 코너에 맞췄습니다 이부분 디자인팀에 업데이트 하겠습니다!

serena0720 replied:
10으로 통일하기로 했습니당~ 꼼꼼한 확인👍 갓갓

zekexros commented on Projects/DesignSystem/Sources/Views/SelectPhoto/SelectPhoto.swift:
```diff
@@ -0,0 +1,57 @@
+import SwiftUI
+
+public struct SelectPhoto: View {
+  @Binding private var selectedPhotosInfo: [PhotoInfo]
```
P2:
Binding일 필요가 없어보입니다. selectedPhotosInfo를 전달받아 SelectPhoto 안에서 값을 변경해주는 곳이 없는 것 같아요

serena0720 replied:
오호... 아 맞네요... 와 감사합니다!

zekexros commented on Projects/Features/Coordinator/CreateEventCoordinator/CreateEventCoordinatorCore.swift:
```diff
@@ -0,0 +1,58 @@
+      case .router(.routeAction(id: _, action: .createEventStart(.moveToGroupMemberList))):
+        state.routes.push(.moveToGroupMemberList(.init()))
+        
+      case .router(.routeAction(id: _, action: .moveToGroupMemberList(.moveBackToEvent))):
+        state.routes.pop()
+        
+      case .router(.routeAction(id: _, action: .moveToCreateEventProcess(.moveToEventStart))):
+        state.routes.pop()
+        
+      case .router(.routeAction(id: _, action: .moveToCreateEventProcess(.moveToGroupListWithEvent))):
+        state.routes.push(.moveToGroupList(.init()))
```
P1:
...? move가 두 번씩 쓰였어요

serena0720 replied:
ㅠ 해당 코디네이터 삭제해야해서 전체적으로 수정 후 재확인 부탁드립니다ㅜ

zekexros commented on Projects/Features/Coordinator/CreateEventCoordinator/CreateEventScreen.swift:
```diff
@@ -0,0 +1,21 @@
+@Reducer(state: .equatable)
+public enum CreateEventScreen {
+  case createEventStart(CreateEventStartCore)
+  case moveToGroupList(GroupListCore)
+  case moveToGroupMemberList(MemberListCore)
+  case moveToCreateEventProcess(CreateEventProcessCore)
```
P1:
첫 번째 케이스(createEventStart) 처럼 나머지 케이스들도 Reducer에 맞춰 작성해주시면 됩니다

serena0720 replied:
꼼꼼한 리뷰 감사합니다!

zekexros commented on Projects/Features/Coordinator/MainCoordinator/MainCoordinatorCore.swift:
```diff
@@ -41,6 +41,9 @@ public struct MainCoordinatorCore {
       case .router(.routeAction(id: _, action: .groupList(.moveToJoinGroup))):
         state.routes.push(.joinGroup(.init()))
         
+      case .router(.routeAction(id: _, action: .groupList(.moveToEvent))):
+        state.routes.presentCover(.createEventCoordinator(.init(routes: [.root(.createEventStart(.init()), embedInNavigationView: true)])))
+        
```
Good👍

zekexros commented on Projects/Features/Coordinator/Project.swift:
```diff
@@ -38,6 +39,19 @@ let project = Project.make(
         .external(externalDependency: .tcaCoordinators),
       ]
+    ),
+    .make(
+      name: "CreateEventCoordinator",
+      product: .staticLibrary,
+      bundleId: "com.mashup.gabbangzip.createEventCoordinator",
+      sources: ["CreateEventCoordinator/**"],
+      dependencies: [
+        .project(target: .createEvent, projectPath: .scene),
+        .project(target: .groupDetail, projectPath: .scene),
+        .project(target: .main, projectPath: .scene),
+        .external(externalDependency: .composableArchitecture),
+        .external(externalDependency: .tcaCoordinators),
+      ]
```
P1:
해당 코디네이터는 groupDetail, main을 알 필요가 없으므로 Dependencies쪽 수정되어야 하겠죠

zekexros reviewed (changes requested):
이벤트를 만든 후 그룹 상세 화면으로 가고 싶다면 `MainCoordinator`에서 조작을 해야 합니다. 그렇다면 이벤트 생성 코디네이터를 닫은 후(dismiss) 그룹 상세로 이동(push)이 되어야 하겠죠. 2번의 화면 전환을 발생시키려면 TCACoordinator의 [Making complex navigation updates](https://github.com/johnpatrickmorgan/TCACoordinators#making-complex-navigation-updates)를 참조하셔서 MainCoordinator에서 구현하시면 됩니다.

기능에 대한 구현 **속도**와 **품질**이 예전에 비해 크게 향상된 것을 보고 놀랐습니다. 감동이에요 👍

zekexros committed:
[Feat] 홈 화면 리팩토링 및 API연동 (#112)

* fix: API 변경된 사항 반영

* chore: API Response 모델 추가

* feat: PushNotificationAPI 구현

* feat: GroupList 리팩토링

* feat: 유틸 파일 추가

* chore: 불필요한 파일 제거 및 프로토콜 채택

* feat: Lovebug 타겟 추가

* fix: typo

* fix: import순서 및 연산 프로퍼티 활용

* fix: Codable -> Decodable

* fix: GroupData 프토로콜 변경

* fix: 불필요한 어노테이션 제거

* refactor: 반복문 리팩토링

* chore: enum raw value 제거

* feat: frame extensions 추가

* fix: PhotoWithFrame 파라미터 타입 변경

* refactor: GroupView 분리

serena0720 committed:
Merge branch 'feat/#43' of https://github.com/mash-up-kr/gabbangzip-iOS into feat/#43

# Conflicts:
#	Projects/Features/Scene/MainScene/GroupList/GroupListCore.swift
#	Projects/Features/Scene/MainScene/GroupList/GroupListView.swift
#	Projects/Features/Scene/Project.swift
#	Tuist/ProjectDescriptionHelpers/TargetDependency+extensions.swift

hope1053 committed:
[Feat] 멤버 리스트 API 연동 (#109)

* chore: MemberList model 정의

* chore: GetMemberList API 메서드 구현

* fix: MemberListView에서 사용하는 모델 수정

* feat: MemberList APi 호출 로직 구현

* feat: MemberList backButton routing action 추가

* design: 링크 복사 → 코드 복사로 타이틀 및 네이밍 변경

* feat: 멤버 정원에 따른 UI 반영

* correct: indentation 수정

* remove: 불필요한 catch문 삭제

* fix: forEach문 식별자 수정

* fix: Text foregroundColor 명시

* correct: 그룹원 초대 문구 수정

serena0720 committed:
chore: Image asset single scale로 수정

serena0720 committed:
chore: Image asset single scale로 수정

serena0720 committed:
chore: 인덴트 수정

serena0720 committed:
chore: 인덴트 수정

serena0720 committed:
chore: 개행 수정 및 조건문 수정

serena0720 committed:
chore: 개행 수정 및 조건문 수정

serena0720 committed:
refactor: SelectPhoto 불필요한 HStack 삭제

serena0720 committed:
refactor: SelectPhoto 불필요한 HStack 삭제

serena0720 committed:
chore: CreateProcessCore 조건문 수정

serena0720 committed:
chore: CreateProcessCore 조건문 수정

hope1053 committed:
[Feat] 그룹 상세 API 연동 1탄 (#119)

* chore: GroupDetail Response 모델 생성

chore: GroupDetail Response 모델 생성

* feat: GroupDetail API 구현

# Conflicts:
#	Projects/Core/Services/PicAPIClients/GroupAPI/GroupAPI.swift
#	Projects/Core/Services/PicAPIClients/GroupAPI/GroupAPIClient.swift

* rename: EventItemView 네이밍 HistoryItemView로 변경

* rename: EventGridView 네이밍 HIstoryGridView로 변경

* fix: 사용하는 모델 GroupDetail → GroupDetailInfo로 교체

* fix: 디자인 컴포넌트 수정 및 사용하지 않는 모델 삭제

# Conflicts:
#	Projects/Features/Scene/GroupDetailScene/MemberList/MemberListCore.swift
#	Projects/Features/Scene/GroupDetailScene/MemberList/MemberListView.swift
#	Projects/Features/Scene/GroupDetailScene/MemberList/MemberView/MemberView.swift

* rename: EventDetail → HistoryDetail로 네이밍 변경

rename: EventDetail → HistoryDetail로 네이밍 변경

* chore: mock data 수정

* feat: routing action 추가

* correct: mock data 변경

* feat: EventAPI, Client 생성

feat: EventAPI, Client 생성

* feat: 현재 그룹 상태가 eventCompleted인 경우 API 호출 구현

* feat: 쿡 찌르기 API 연동

feat: 쿡 찌르기 API 연동

* fix: 중복되는 struct 네이밍 수정

* remove: 불필요한 catch문 삭제

* feat: eventControlButton 삭제 및 EventProgressView 수정

* fix: action 수정

* correct: share image 수정

correct: share image 수정

* feat: EventCompletedView 구현

* feat: EventCompletedView 구현

* remove: 중복되는 파일 삭제

* refactor: PhotoWithFrame 적용

* feat: EventCompletedView PhotoCardView 구현

* fix: deprecated된 메서드 수정

* feat: 이미지 업로드 기능 구현

* design: foregroundColor 수정

* feat: noCurrentEvent인 경우 화면 작업

* refactor: EventCompletedState initializer 수정

* fix: 불필요한 인자 삭제

* fix: noCurrentEvent인 경우에도 eventCompletedView 보이도록 수정

* feat: 이미지 업로드 결과에 따라 토스트 구현

* refactor: EventCompletedCore 삭제 후 GroupDetailCore로 통합

refactor: EventCompletedCore 삭제 후 GroupDetailCore로 통합

* fix: 컨플릭 해결 과정에서 생긴 오류 수정

* fix: PushNotificationAPI로 통일

* remove: 불필요한 프로퍼티 삭제

* refactor: buttonType state에서 관리하도록 수정

* refactor: buttonType state에서 관리하도록 수정

* refactor: Image 로딩해오는 부분 코드 보완

* fix: 리뷰 반영

hope1053 committed:
[Feat] 그룹 상세 API 연동 2탄 (#121)

* feat: HistoryGridView 구현

# Conflicts:
#	Projects/Features/Scene/GroupDetailScene/GroupDetail/HistoryGrid/HistoryItemView.swift

* feat: HistoryDetailView 구현

# Conflicts:
#	Projects/Features/Scene/GroupDetailScene/HistoryDetail/HistoryDetailCore.swift

* feat: GroupDetailCoordinator 생성

* feat: GroupList → GroupDetail 이동 로직 구현

# Conflicts:
#	Projects/Features/Scene/CreateGroupScene/CreateGroupCompletion/CreateGroupCompletionView.swift

* feat: HistoryView tap 이벤트 구현

* feat: Coordinator 작업

* feat: 이미지 공유 기능 구현

* chore: mock data 수정

* fix: GroupDetail present → push로 변경

* fix: GroupDetailInfo mock → optional 타입으로 변경

* fix: 누락된 action 수정

* fix: MemberList mock → optional 타입으로 변경

zekexros committed:
[Fix] GroupList 이름 변경 및 Response 오타 수정 (#123)

* fix: FCMToken 오타 수정

* fix: GroupList -> Home 수정

serena0720 committed:
refactor: CreateEventProcess 날짜 DateFormatter및 RecentEvent로 수정

serena0720 committed:
refactor: CreateEventProcess 날짜 DateFormatter및 RecentEvent로 수정

serena0720 committed:
refactor: SelectPhotosInfo 타입 변경 및 컨벤션 수정

serena0720 committed:
refactor: SelectPhotosInfo 타입 변경 및 컨벤션 수정

serena0720 committed:
refactor: CreateEventCoordinator -> CreateEvent 로 수정

CreateEvent 화면이 단일 화면으로 수정되면서 전면 수정

serena0720 committed:
refactor: CreateEventCoordinator -> CreateEvent 로 수정

CreateEvent 화면이 단일 화면으로 수정되면서 전면 수정

serena0720 committed:
fix: Coordinator 부분 추가 수정

serena0720 committed:
fix: Coordinator 부분 추가 수정

serena0720 committed:
Merge branch 'feat/#43' of https://github.com/mash-up-kr/gabbangzip-iOS into feat/#43

# Conflicts:
#	Projects/Features/Coordinator/MainCoordinator/MainCoordinatorCore.swift
#	Projects/Features/Scene/GroupDetailScene/MemberList/MemberListCore.swift

serena0720 commented:
## 📢 Notes
디자인 수정으로 CreateEventCoordinator가 아닌 CreateEvent로 수정하였습니다.
이부분 참고 부탁드립니다.
@GREENOVER @zekexros @hope1053

serena0720 committed:
fix: 이벤트 생성버튼 터치 시 이벤트 생성 페이지로 이동, CreateEvent로 네이밍 변경, CreateEventView 분리

serena0720 committed:
refactor: PhotoInfo의 Equatable 삭제

zekexros commented on Projects/Features/Coordinator/MainCoordinator/MainCoordinatorCore.swift:
```diff
@@ -53,6 +56,12 @@ public struct MainCoordinatorCore {
+        
+      case .router(.routeAction(id: _, action: .createEvent(.moveToGroupListWithEvent))):
+        state.routes.pop()
+        
```
P1:
`GroupList`가 `Home`으로 바뀌었습니다

serena0720 committed:
chore: GroupList -> Home 네이밍 수정
